packages/functional-tests/src/runner/test-runner.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import 'reflect-metadata';
+
+import { inject } from 'inversify';
+import { Logger, LogLevel } from 'logger';
+import { LogSource, TestDefinition, TestEnvironment, TestResult } from '../common-types';
+import { getDefinedTestsMetadata } from '../test-decorator';
+
+// tslint:disable: no-any no-unsafe-any
+
+export interface TestRunLogProperties {
+    source: LogSource;
+    testContainer: string;
+    testName: string;
+    environment: string;
+    result: TestResult;
+    error?: string;
+}
+
+export interface TestContainerLogProperties {
+    source: LogSource;
+    testContainer: string;
+    environment: string;
+    result: TestResult;
+}
+
+export class TestRunner {
+    public constructor(@inject(Logger) private readonly logger: Logger) {}
+
+    public async run(testContainer: object, env: TestEnvironment): Promise<void> {
+        const definedTests = getDefinedTestsMetadata(testContainer);
+        // tslint:disable-next-line: no-bitwise
+        const targetedTests = definedTests.filter(definedTest => definedTest.environments & env);
```

Comment:
NIT: it would be nice to extract the bit manipulation to a separate function to make it more clear how we are filtering here.

Reply:
will update next pr